File: biblioteca_reservas_libros/src/java/Reserva.java
package java;
import java.util.Date;

/**
 * La clase Reserva representa una reserva de un libro realizada por un usuario en la biblioteca.
 */
public class Reserva {
    private String usuario;
    private Libro libro;
    private Date fechaReserva;
    private Date fechaDevolucion;

    /**
     * Constructor de la clase Reserva.
     *
     * @param usuario         El nombre del usuario que realiza la reserva.
     * @param libro           El libro que se reserva.
     * @param fechaReserva    La fecha en que se realiza la reserva.
     * @param fechaDevolucion La fecha de devolución prevista del libro.
     */
    public Reserva(String usuario, Libro libro, Date fechaReserva, Date fechaDevolucion) {
        this.usuario = usuario;
        this.libro = libro;
        this.fechaReserva = fechaReserva;
        this.fechaDevolucion = fechaDevolucion;
    }

    /**
     * Obtiene el nombre del usuario que realizó la reserva.
     *
     * @return El nombre del usuario.
     */
    public String getUsuario() {
        return usuario;
    }

    /**
     * Obtiene el libro que se reservó.
     *
     * @return El libro reservado.
     */
    public Libro getLibro() {
        return libro;
    }

    /**
     * Obtiene la fecha en que se realizó la reserva.
     *
     * @return La fecha de reserva.
     */
    public Date getFechaReserva() {
        return fechaReserva;
    }

    /**
     * Obtiene la fecha de devolución prevista del libro.
     *
     * @return La fecha de devolución prevista.
     */
    public Date getFechaDevolucion() {
        return fechaDevolucion;
    }
}

File: biblioteca_reservas_libros/src/js/Reserva.js
/**
 * Clase que representa una reserva de un libro en una biblioteca.
 * @class
 */
class Reserva {
    /**
     * Crea una instancia de la clase Reserva.
     * @constructor
     * @param {string} usuario - El nombre de usuario del cliente que realiza la reserva.
     * @param {Libro} libro - El libro que se ha reservado.
     * @param {Date} fechaReserva - La fecha en la que se realizó la reserva.
     * @param {Date} fechaDevolucion - La fecha de devolución prevista del libro reservado.
     */
    constructor(usuario, libro, fechaReserva, fechaDevolucion) {
      /**
       * El nombre de usuario del cliente que realiza la reserva.
       * @type {string}
       */
      this.usuario = usuario;
  
      /**
       * El libro que se ha reservado.
       * @type {Libro}
       */
      this.libro = libro;
  
      /**
       * La fecha en la que se realizó la reserva.
       * @type {Date}
       */
      this.fechaReserva = fechaReserva;
  
      /**
       * La fecha de devolución prevista del libro reservado.
       * @type {Date}
       */
      this.fechaDevolucion = fechaDevolucion;
    }
  
    /**
     * Obtiene el nombre de usuario del cliente que realizó la reserva.
     * @returns {string} - El nombre de usuario del cliente.
     */
    getUsuario() {
      return this.usuario;
    }
  
    /**
     * Obtiene el libro que se ha reservado.
     * @returns {Libro} - El libro reservado.
     */
    getLibro() {
      return this.libro;
    }
  
    /**
     * Obtiene la fecha en la que se realizó la reserva.
     * @returns {Date} - La fecha de reserva.
     */
    getFechaReserva() {
      return this.fechaReserva;
    }
  
    /**
     * Obtiene la fecha de devolución prevista del libro reservado.
     * @returns {Date} - La fecha de devolución prevista.
     */
    getFechaDevolucion() {
      return this.fechaDevolucion;
    }
  }